index.js

```diff
@@ -22,11 +33,11 @@ function fastifyRequestContext(fastify, opts, next) {
         ? opts.defaultStoreValues()
         : opts.defaultStoreValues
 
-    als.runWith(() => {
+    asyncLocalStorage.run(defaultStoreValues || {}, () => {
```

Comment:
Just for my curiosity and because I have no knowledge regarding the AsyncLocalStorage functionality.

Wouldnt it be safer regarding prototype pollution to have a Null Object as Store? 

like 
```js
const AsyncStore = function AsyncStore () { }
AsyncStore.prototype = Object.create(null)
```
And then use it like this:
```
asyncLocalStorage.run(defaultStoreValues || new AsyncStore(), () => {
```
?

Also: If defaultStoreValues is an Object, wouldnt it be used by reference? So, wouldnt the object be shared between the instances?

Reply:
ALS ensures that each request gets their own instance, that's its entire thing.

I'm not sure we need to protect against prototype pollution, store manipulation logic is entirely controlled by the app here. Do you see a potential attack vector somewhere?

Reply:
I have no idea. The example on node documentation is using a primitive datatype as store. We are injecting an object as store from outside the ALS. So i have the feeling, that it could be by referenced. And so prototype pollution could be possible. 

But like I said: i have no idea. I just got the idea, because the first attempt was a Map and then changed to an Object.

Reply:
I changed this to:

```js
asyncLocalStorage.run({ ...defaultStoreValues }, () => {
```

That way a new context object is always created for every new request, which I think is what is expected, as @Uzlopak says.

Regarding having an `Object.create(null)` as the base, I don't see the point of that here actually, if anyone is assigning unsanitized user input to the top-level keys of the request context, then they are doing something really bad.

Reply:
let's add a test which validates that we are not leaking mutated defaultValues across requests then